src/components/tl/sharp/tl_sharp_team.c

```diff
@@ -30,15 +32,25 @@ UCC_CLASS_INIT_FUNC(ucc_tl_sharp_team_t, ucc_base_context_t *tl_context,
     self->sharp_context = NULL;
     self->rcache        = NULL;
     self->oob_ctx.ctx   = UCC_TL_TEAM_CTX(self);
+
+    set.myrank = UCC_TL_TEAM_RANK(self);
+    set.map    = UCC_TL_TEAM_MAP(self);
+
     if (UCC_TL_SHARP_TEAM_LIB(self)->cfg.use_internal_oob) {
-        self->oob_ctx.subset.map    = UCC_TL_TEAM_MAP(self);
-        self->oob_ctx.subset.myrank = UCC_TL_TEAM_RANK(self);
+        self->oob_ctx.subset = set;
     } else {
         self->oob_ctx.oob = &UCC_TL_TEAM_OOB(self);
     }
 
     if (sharp_ctx == NULL) {
-        status = ucc_tl_sharp_context_init(ctx, &self->sharp_context, &self->oob_ctx);
+        status = ucc_topo_init(set, ctx->super.super.ucc_context->topo, &topo);
```

Comment:
can be moved to ucc_tl_sharp_context_init, identical for both cases when we have comm per ctx and comm per team